forge-gui/res/cardsfolder/upcoming/thran_portal.txt

```diff
@@ -0,0 +1,12 @@
+Name:Thran Portal
+ManaCost:no cost
+Types:Land Gate
+K:ETBReplacement:Other:LandTapped
+SVar:LandTapped:DB$ Tap | Defined$ Self | ETB$ True | ConditionCheckSVar$ ETBCheckSVar2 | ConditionSVarCompare$ GT2 | SpellDescription$ CARDNAME enters the battlefield tapped unless you control two or fewer other lands.
+K:ETBReplacement:Other:DBChooseBasic
+SVar:DBChooseBasic:DB$ ChooseType | Type$ Basic Land | SpellDescription$ As CARDNAME enters the battlefield, choose a basic land type.
+S:Mode$ Continuous | Affected$ Card.Self | AddType$ ChosenType | Description$ CARDNAME is the chosen type in addition to its other types.
+S:Mode$ RaiseCost | Type$ Ability | ValidCard$ Card.Self | Cost$ PayLife<1> | EffectZone$ All | Description$ Activated abilities of CARDNAME cost an additional "1 life" to activate.
```

Comment:
It should be mana abilities only

Reply:
I think this required some engine changes, interested what you think about this